src/Ashampoo.Translation.Systems.Formats/src/CSV/CsvFormat.cs

```diff
@@ -0,0 +1,209 @@
+        await WriteHeader(writer);
+        var csvWriter = new CsvWriter(writer, new CsvConfiguration(CultureInfo.InvariantCulture)
+        {
+            Delimiter = Delimiter,
+            DetectDelimiter = false,
+            TrimOptions = TrimOptions.Trim,
+        });
+        csvWriter.WriteHeader<CsvRecordFormat>();
+        await csvWriter.NextRecordAsync();
+        foreach (var translationUnit in TranslationUnits)
+        {
+            foreach (var translation in translationUnit.Translations.Where(t =>
+                         t.Language.Value == Header.TargetLanguage.Value))
+            {
+                await WriteRow(translationUnit, translation, csvWriter);
+            }
+        }
+
+        await csvWriter.FlushAsync();
+        await writer.FlushAsync();
+    }
+
+    private async Task WriteRow(ITranslationUnit unit, ITranslation translation, CsvWriter writer)
+    {
+        var sourceTranslation =
+            unit.Translations.FirstOrDefault(t => t.Language == Header.SourceLanguage);
+        var record = new CsvRecordFormat
+        {
+            Id = unit.Id,
+            Original = sourceTranslation?.Value ?? string.Empty,
+            Translation = translation.Value,
+            Comments = string.Join("|", translation.Comments)
+        };
+        writer.WriteRecord(record);
+        await writer.NextRecordAsync();
+    }
+
+    private async Task WriteHeader(TextWriter writer)
+    {
+        await CsvFormatHeader.WriteHeader(writer);
+        await writer.WriteLineAsync();
+    }
+
+    // TODO: Can this be made Protected in a abstract class to avoid duplicate code?
+    private async Task<bool> ConfigureOptionsAsync(FormatReadOptions? options)
+    {
+        if (string.IsNullOrWhiteSpace(options?.TargetLanguage.Value))
+        {
+            Guard.IsNotNull(options?.FormatOptionsCallback);
+
+            FormatStringOption targetLanguageOption = new("Target language", true);
+            FormatOptions formatOptions = new()
+            {
+                Options =
+                [
+                    targetLanguageOption
+                ]
+            };
+
+            await options.FormatOptionsCallback.Invoke(formatOptions); // Invoke callback
+            if (formatOptions.IsCanceled) return false;
+
+            Header.TargetLanguage = Language.Parse(targetLanguageOption.Value);
+        }
+        else
+        {
+            Header.TargetLanguage = options.TargetLanguage;
+        }
+
+        Header.SourceLanguage = options.SourceLanguage;
+        return true;
+    }
+
+    private void ApplyDelimiter()
+    {
+        if (!Header.AdditionalHeaders.TryGetValue("delimiter", out var delimiter)) return;
+        var actualDelimiter = delimiter.ToCharArray().FirstOrDefault();
+        if (actualDelimiter is ',' or ';' or '|')
+        {
+            CsvFormatHeader.Delimiter = actualDelimiter;
+        }
+    }
+}
+
+internal record CsvRecordFormat
```

Comment:
Der record kann glaube ich in den file scope verschoben werden (file record CsvRecord)